Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp

```diff
@@ -0,0 +1,419 @@
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(0)*std::sin(p.rdata(PIdx::theta)); });
+        ParallelDescriptor::ReduceRealMin(ymin);
+#elif (defined WARPX_DIM_XZ)
+        Real ymin = 0.0_rt;
+#else
+        Real ymin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(1); });
+        ParallelDescriptor::ReduceRealMin(ymin);
+#endif
+
+        // ymax
+#if (defined WARPX_DIM_RZ)
+        Real ymax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(0)*std::sin(p.rdata(PIdx::theta)); });
+        ParallelDescriptor::ReduceRealMax(ymax);
+#elif (defined WARPX_DIM_XZ)
+        Real ymax = 0.0_rt;
+#else
+        Real ymax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(1); });
+        ParallelDescriptor::ReduceRealMax(ymax);
+#endif
+
+        // zmin
+        Real zmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(index_z); });
+        ParallelDescriptor::ReduceRealMin(zmin);
+
+        // zmax
+        Real zmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(index_z); });
+        ParallelDescriptor::ReduceRealMax(zmax);
+
+        // uxmin
+        Real uxmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::ux); });
+        ParallelDescriptor::ReduceRealMin(uxmin);
+
+        // uxmax
+        Real uxmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::ux); });
+        ParallelDescriptor::ReduceRealMax(uxmax);
+
+        // uymin
+        Real uymin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uy); });
+        ParallelDescriptor::ReduceRealMin(uymin);
+
+        // uymax
+        Real uymax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uy); });
+        ParallelDescriptor::ReduceRealMax(uymax);
+
+        // uzmin
+        Real uzmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uz); });
+        ParallelDescriptor::ReduceRealMin(uzmin);
+
+        // uzmax
+        Real uzmax = ReduceMax( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uz); });
+        ParallelDescriptor::ReduceRealMax(uzmax);
+
+        // gmin
+        Real gmin = ReduceMin( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            Real ux = p.rdata(PIdx::ux);
+            Real uy = p.rdata(PIdx::uy);
+            Real uz = p.rdata(PIdx::uz);
+            Real us = ux*ux + uy*uy + uz*uz;
+            return std::sqrt(1.0 + us*inv_c2);
```

Comment:
This should be different for photons, since their mass is zero. Quite often people define a sort of `gamma` factor for photons, which is simply their momentum normalized with respect to m_e c.
You could simply replace `std::sqrt(1.0 + us*inv_c2);` with `std::sqrt(us*inv_c2);` for photons